Applications/Utils/MeshEdit/UnityPreprocessing.cpp

```diff
@@ -0,0 +1,245 @@
+            return true;
+    }
+    return false;
+}
+
+bool isCellVec(MeshLib::Properties const& props, std::string const& name)
+{
+    if (checkType<int>(props, name))
+        return true;
+    if (checkType<double>(props, name))
+        return true;
+    if (checkType<long>(props, name))
+        return true;
+    return false;
+}
+
+bool containsCellVecs(MeshLib::Mesh const& mesh)
+{
+    MeshLib::Properties const& props (mesh.getProperties());
+    std::vector<std::string> const& vec_names(props.getPropertyVectorNames());
+    for (std::string name : vec_names)
+    {
+        if (isCellVec(props, name))
+            return true;
+    }
+    return false;
+}
+
+template <class T>
+MeshLib::Element* createElement(MeshLib::Element const& e,
+    std::vector<MeshLib::Node*> &nodes,
+    std::vector<std::vector<std::size_t>> &node_map)
+{
+    std::size_t const n_nodes = e.getNumberOfNodes();
+    MeshLib::Node** new_nodes = new MeshLib::Node*[n_nodes];
+    for (unsigned i = 0; i<n_nodes; ++i)
+    {
+        new_nodes[i] = new MeshLib::Node(e.getNode(i)->getCoords(), nodes.size());
+        node_map[e.getNodeIndex(i)].push_back(nodes.size());
+        nodes.push_back(new_nodes[i]);
+    }
+    return new T(new_nodes);
+}
+
+template <class T>
+bool fillPropVec(MeshLib::Properties const& props,
+    std::string const& name,
+    MeshLib::Properties &new_props,
+    std::vector<MeshLib::Element*> const& elems,
+    std::vector<std::vector<std::size_t>> const& node_map,
+    std::size_t total_nodes)
+{
+    if (!props.existsPropertyVector<T>(name))
+        return false;
+
+    MeshLib::PropertyVector<T> const*const vec = props.getPropertyVector<T>(name);
+    if (vec->getNumberOfComponents() != 1)
+    {
+        INFO("Ignoring array \"%s\" (more than one component).", name.c_str());
+        return false;
+    }
+
+    MeshLib::PropertyVector<T>* new_vec =
+        new_props.createNewPropertyVector<T>(vec->getPropertyName(), MeshLib::MeshItemType::Node, 1);
+    new_vec->resize(total_nodes);
+    if (vec->getMeshItemType() == MeshLib::MeshItemType::Node)
+    {
+        INFO("Migrating node array \"%s\" to new mesh structure...", name.c_str());
+        std::size_t const n_nodes (node_map.size());
+        for (std::size_t i = 0; i<n_nodes; ++i)
+        {
+            std::size_t const n_nodes = node_map[i].size();
+            for (std::size_t j = 0; j<n_nodes; ++j)
+                (*new_vec)[node_map[i][j]] = (*vec)[i];
+        }
+    }
+    else if (vec->getMeshItemType() == MeshLib::MeshItemType::Cell)
+    {
+        INFO("Transforming cell array \"%s\" into node array...", name.c_str());
+        std::size_t const n_elems (vec->size());
+        for (std::size_t i = 0; i<n_elems; ++i)
+        {
+            std::size_t const n_nodes = elems[i]->getNumberOfNodes();
+            for (std::size_t j = 0; j<n_nodes; ++j)
+                (*new_vec)[elems[i]->getNodeIndex(j)] = (*vec)[i];
```

Comment:
Does that mean, that you write all cell data to their adjacent nodes?
Since there is no averaging done, does that mean that the last cell value just wins?